### PR TITLE
Filter categories

### PR DESCRIPTION
**Pull request by AtulPant2704**
- Display videos in Video Listing page according to category selected
- Refactored Folder structure

Preview:- https://deploy-preview-10--checkmate-stream.netlify.app/explore

`Note:- Review this PR by commits, last two commits are just change in folder structure and file naming, no functionality change, so you can skip that.`

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -13,17 +13,19 @@ ReactDOM.render(
...
+                      <App />
+                    </HistoryProvider>
+                  </WatchLaterProvider>
+                </LikesProvider>
+              </PlaylistsProvider>
+            </PlaylistModalProvider>
+          </CategoryProvider>
         </VideosProvider>
       </AuthProvider>
     </Router>
```
Comment (VishalPatil18) [inline src/index.js]: Your `Router` should be the innermost child so that it has access to all the providers. It is a good practice to follow.

**Review by VishalPatil18 — APPROVED** [review]
Just one suggestion was added. The rest looks quite good. Great work @AtulPant2704 👍

**Review by mohammed-sas — APPROVED** [review]
1. filter functionalities are working as intended
2. category service and handler has been coded well
3. code is clean and modular

**Review by AnkurChunekar — APPROVED** [review]
- Categories functionality works as fine.
- Code is neat and clean.
- Great work @AtulPant2704

**Pull request merged**